app/Plugin.php

```diff
@@ -170,51 +171,58 @@ public function getAccessPoints(): array {
         return $accesspoints;
     }
 
-    public function getScopes(): array {
-        $info = self::getInfo(base_path("app/Plugins/$this->name"), false);
-        $scopes = [];
-        if($info !== false) {
-            if(array_key_exists('scopes', $info)) {
-                foreach($info['scopes'] as $scope) {
-                    $attributes = $scope['@attributes'];
-                    if(!array_key_exists('src', $attributes)) {
-                        Log::error('<scope> attribute \'src\' is required');
-                        continue;
-                    }
-                    if(!array_key_exists('on', $attributes)) {
-                        Log::error('<scope> attribute \'on\' is required');
-                        continue;
-                    }
-
-                    $src = $attributes['src'];
-                    $on = $attributes['on'];
-
-                    $srcDir = base_path("app/Plugins/$this->name/Scopes");
-                    if(!file_exists($srcDir) || !is_dir($srcDir)) {
-                        Log::error('Missing \'Scopes\' directory');
-                        continue;
-                    }
-                    $srcPath = "{$srcDir}/{$src}";
-                    if(!file_exists($srcPath)) {
-                        Log::error("Missing file '$src'");
-                        continue;
-                    }
-                    if(!class_exists($on)) {
-                        Log::error("Class '{$on}' does not exist!");
-                        continue;
-                    }
-                    $className = Str::replaceEnd('.php', '', $src);
-                    $namespacedSrc = "App\Plugins\\$this->name\Scopes\\$className";
+    private function getScopeCacheKey(): string {
+        return 'plugin_scopes_' . $this->id;
+    }
 
-                    if(!array_key_exists($on, $scopes)) {
-                        $scopes[$on] = [];
+    public function getScopes(): array {
+        return Cache::rememberForever($this->getScopeCacheKey(), function() {
```

Comment:
I added the Cache to make access faster without the need to parse the info.xml on every request.
It will be invalidated on install, update, uninstall and remove of the respective plugin.